module/VuFind/src/VuFind/Cache/Manager.php

```diff
@@ -256,6 +292,27 @@ public function addLanguageCacheForTheme($themeName)
         return $cacheName;
     }
 
+    /**
+     * Ensure that a file cache is properly set up
+     *
+     * @param string $name Cache name
+     *
+     * @return void
+     */
+    protected function ensureFileCache(string $name): void
+    {
+        if ($this->ensuredCached[$name] ?? false) {
```

Comment:
Do we need this separate ensuredCached property? Is there a reason we can't just check if $this->caches[$name] is set?

I ask this for two reasons:

1.) I don't like the name of the ensuredCached property, but if we don't need the property at all, we don't need to discuss what to call it.

2.) I don't think this plays well with the addDownloaderCache method, since that creates a cache without setting the ensuredCached value. That leads me to think that if we need a flag at all, we should probably set it inside the createFileCache method and not inside ensureFileCache.

Reply:
`$this->caches` has the namespaced keys as the names. We can of course parse the names from the keys, but it isn't very clean. That's why I added a separate property to track created file caches. Perhaps tracking could be a flag in the cacheSpecs array instead?

addDownloaderCache hasn't been part of the pre-ensured caches, so I left it out from these changes as well. If you'd rather have it in, I can add it.

Reply:
Ahh, yes. But $this->cacheSettings seems to have the right level of granularity.

What if you replace the ensuredCached check with `if (!isset($this->cacheSettings[$name]) && isset($this->cacheSpecs[$name])) {` ?

Reply:
Or, I guess, more appropriately, get rid of all the ensuredCached logic and just change `if ($config = $this->cacheSpecs[$name] ?? null) {` to `if (!isset($this->cacheSettings[$name]) && $config = $this->cacheSpecs[$name] ?? null) {`.

Reply:
Changed!